notebooks/Credit Loan Example.ipynb

```diff
@@ -0,0 +1,1861 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def model_performance_sweep(models_dict, X_test, y_test, A_test):\n",
+    "    \"\"\"\n",
+    "    Computes the equalized_odds_difference and balanced_error_rate for a given list of inner models\n",
+    "    learned by the ExponentiatedGradient algorithm. Returns a DataFrame containing the epsilon level of the model,\n",
+    "    the index of the model, the equalized_odds_difference score and the balanced_error for the model.\n",
+    "    \n",
+    "    Parameters:\n",
+    "    - models_dict: Dictionary mapping model ids to a model.\n",
+    "    - X_test: The data features of the testing data set\n",
+    "    - y_test: The target labels of the teting data set\n",
+    "    - A_test: The sensitive feature of the testing data set.\n",
+    "    \n",
+    "    Returns:\n",
+    "        DataFrame where each row represents a model (epsilon, index) and its performance metrics\n",
+    "    \"\"\"\n",
+    "    performances = []\n",
+    "    for (eps, models) in models_dict.items():\n",
+    "        eq_odds_difference = aggregate_predictor_performances(models, equalized_odds_difference, X_test, y_test, A_test)\n",
+    "        bal_acc_score = aggregate_predictor_performances(models, balanced_accuracy_score, X_test, y_test)\n",
+    "        for (i, score) in enumerate(eq_odds_difference):\n",
+    "            performances.append((\n",
+    "                eps,\n",
+    "                i,\n",
+    "                score,\n",
+    "                (1-bal_acc_score[i])\n",
+    "            ))\n",
+    "    performances_df = pd.DataFrame.from_records(performances,\n",
+    "                                               columns=[\"epsilon\", \"index\", \"equalized_odds\", \"balanced_error\"])    \n",
+    "    return performances_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "performance_df = model_performance_sweep(all_models, X_test, y_test, A_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "performance_subset = performance_df.loc[:, [\"equalized_odds\", \"balanced_error\"]]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask, pareto_subset = filter_dominated_rows(performance_subset)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "performance_df_masked = performance_df.loc[mask, :]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's plot the performance trade-offs between all of our models. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for index, row in performance_df_masked.iterrows():\n",
```

Comment:
change the label to something like "eps=0.01, idx=10"

Reply:
Done. Some of the labels overlap now, but I don't think it's a big deal. Most of them are still readable.